### ISSUE TITLE
[JENKINS-22641] Jenkins no longer kills running processes after job fails

### ISSUE THREAD

**Issue by jenkins-infra-bot** [timja-org/jenkins-gh-issues-poc-11-23#11687]
<p>Starting at version 1.553, Jenkins no longer seems to kill running processes after a build failure.</p>

<p>We have several jobs that start a Tomcat instance and run various end-to-end-tests; if the build fails Jenkins doesn't execute the shutdown scripts and we rely on the process killer to clean up the Tomcat instance.</p>

<p>This can be duplicated more easily by creating a free-form job and adding two shell scripts, the first that starts a simple command such as "nohup sleep 10000 &" and the second "/bin/false".  After the job exits the sleep process is still running.  Prior to version 1.553, it would be killed.</p>

<p>There are no log messages to indicate a problem.</p>

<p>I can reproduce this on CentOS 6, Red Hat EL 5 and Red Hat EL 4, both with a job running on the local master, and on a slave node.  Also tested with both 32-bit and 64-bit Oracle Java 7 JDKs.</p>

<p>We're using the built-in Winstone container.</p>

---
<details><summary><i>Originally reported by <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/toadnik17.jpg" title="toadnik17's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=toadnik17">toadnik17</a>, imported from: <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-22641" target="_blank">Jenkins no longer kills running processes after job fails</a></i></summary>
<i><ul>
<li><b>assignee</b>: <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/schristou.jpg" title="schristou's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=schristou">schristou</a>
<li><b>status</b>: Closed
<li><b>priority</b>: Critical
<li><b>component(s)</b>: core
<li><b>label(s)</b>: 1.565.2-fixed, regression
<li><b>resolution</b>: Fixed
<li><b>resolved</b>: 2014-12-12T16:21:04+00:00
<li><b>votes</b>: 12
<li><b>watchers</b>: 23
<li><b>imported</b>: 2025-11-23
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p>Starting at version 1.553, Jenkins no longer seems to kill running processes after a build failure.</p>

<p>We have several jobs that start a Tomcat instance and run various end-to-end-tests; if the build fails Jenkins doesn't execute the shutdown scripts and we rely on the process killer to clean up the Tomcat instance.</p>

<p>This can be duplicated more easily by creating a free-form job and adding two shell scripts, the first that starts a simple command such as "nohup sleep 10000 &amp;" and the second "/bin/false".  After the job exits the sleep process is still running.  Prior to version 1.553, it would be killed.</p>

<p>There are no log messages to indicate a problem.</p>

<p>I can reproduce this on CentOS 6, Red Hat EL 5 and Red Hat EL 4, both with a job running on the local master, and on a slave node.  Also tested with both 32-bit and 64-bit Oracle Java 7 JDKs.</p>

<p>We're using the built-in Winstone container.</p></pre>
</details>
</details>
<ul><li><i>environment</i>: <code>Jenkins 1.553 or later on CentOS 6.3 with Oracle Java 7 JDK</code></li></ul>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/danielbeck.png" title="danielbeck's avatar" /> danielbeck:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-22641?focusedId=202313&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-202313">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>I'm trying to find out the cause for the different versions reported here. It would help if you report all of the following in an unambiguous format:</p>

<ul class="alternate" type="square">
	<li>last known good Jenkins version</li>
	<li>first known broken Jenkins version</li>
	<li>OS of the master and whether this issue appears with jobs executed there</li>
	<li>OS of all relevant slaves (i.e. slaves the issue appears, or is known not to appear on), and how they're started (e.g. JNLP, SSH slaves)</li>
	<li>Whether anything is reported in the logs</li>
	<li>The simplest reproducible job the issue appears with. does two shell build steps, one <tt>nohup sleep 10000 &amp;</tt>, the other <tt>/bin/false</tt> do the job? If so, on which nodes? More/fewer than your real issue?</li>
</ul>

</pre>
</details>

</ul>
</details>

<p>I'm trying to find out the cause for the different versions reported here. It would help if you report all of the following in an unambiguous format:</p>

<ul class="alternate" type="square">
	<li>last known good Jenkins version</li>
	<li>first known broken Jenkins version</li>
	<li>OS of the master and whether this issue appears with jobs executed there</li>
	<li>OS of all relevant slaves (i.e. slaves the issue appears, or is known not to appear on), and how they're started (e.g. JNLP, SSH slaves)</li>
	<li>Whether anything is reported in the logs</li>
	<li>The simplest reproducible job the issue appears with. does two shell build steps, one <tt>nohup sleep 10000 &</tt>, the other <tt>/bin/false</tt> do the job? If so, on which nodes? More/fewer than your real issue?</li>
</ul>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/pszilard.jpg" title="pszilard's avatar" /> pszilard:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-22641?focusedId=204265&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-204265">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>We may be experiencing the same bug in out jeknins setup - although I have seen orphaned processes left behind by jenkins before, but now it's become a plague.</p>

<ul>
	<li>last known good Jenkins version:
<b>no idea</b></li>
	<li>first known broken Jenkins version:
<b>1.554.1</b></li>
	<li>OS of the master and whether this issue appears with jobs executed there:
<b>Ubuntu 12.04.4 LTS, does not run jobs</b></li>
	<li>OS of all relevant slaves (i.e. slaves the issue appears, or is known not to appear on), and how they're started (e.g. JNLP, SSH slaves)
<b>Ubuntu 10.04, 12.04, 13.10; OS X 10.7 (SSH slaves)</b></li>
	<li>Whether anything is reported in the logs
<b>could not find anything relevant</b></li>
</ul>


<p>Note that I have the feeling this only happens to aborted jobs, but I can't confirm it ATM.</p>
</pre>
</details>

</ul>
</details>

<p>We may be experiencing the same bug in out jeknins setup - although I have seen orphaned processes left behind by jenkins before, but now it's become a plague.</p>

<ul>
	<li>last known good Jenkins version:<br/>
<b>no idea</b></li>
	<li>first known broken Jenkins version:<br/>
<b>1.554.1</b></li>
	<li>OS of the master and whether this issue appears with jobs executed there:<br/>
<b>Ubuntu 12.04.4 LTS, does not run jobs</b></li>
	<li>OS of all relevant slaves (i.e. slaves the issue appears, or is known not to appear on), and how they're started (e.g. JNLP, SSH slaves)<br/>
<b>Ubuntu 10.04, 12.04, 13.10; OS X 10.7 (SSH slaves)</b></li>
	<li>Whether anything is reported in the logs<br/>
<b>could not find anything relevant</b></li>
</ul>


<p>Note that I have the feeling this only happens to aborted jobs, but I can't confirm it ATM.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/schristou.jpg" title="schristou's avatar" /> schristou:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-22641?focusedId=205814&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-205814">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>I was able to reproduce this issue locally and I am working on a fix.</p>
</pre>
</details>

</ul>
</details>

<p>I was able to reproduce this issue locally and I am working on a fix.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/schristou.jpg" title="schristou's avatar" /> schristou:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-22641?focusedId=205924&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-205924">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>I created a pull request: <a href="https://github.com/jenkinsci/jenkins/pull/1322" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/jenkins/pull/1322</a> to resolve this issue.</p>
</pre>
</details>

</ul>
</details>

<p>I created a pull request: <a href="https://github.com/jenkinsci/jenkins/pull/1322" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/jenkins/pull/1322</a> to resolve this issue.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/schristou.jpg" title="schristou's avatar" /> schristou:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-22641?focusedId=205994&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-205994">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>This looks to be a regression in commons-io library. I logged <a href="https://issues.apache.org/jira/browse/IO-453" class="external-link" target="_blank" rel="nofollow noopener">IO-453</a> to track the changes for the commons-io library.</p>
</pre>
</details>

</ul>
</details>

<p>This looks to be a regression in commons-io library. I logged <a href="https://issues.apache.org/jira/browse/IO-453" class="external-link" target="_blank" rel="nofollow noopener">IO-453</a> to track the changes for the commons-io library.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/danielbeck.png" title="danielbeck's avatar" /> danielbeck:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-22641?focusedId=206075&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-206075">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=schristou" class="user-hover" rel="schristou">schristou</a>: I merged your fix into a private build of 1.554.3 and verified this fixes the issue as described in my earlier comment from 09/May/14 11:42 PM. Unpatched 1.554.3 doesn't kill 'sleep' on the slave, patched does.</p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=schristou" class="user-hover" rel="schristou">schristou</a>: I merged your fix into a private build of 1.554.3 and verified this fixes the issue as described in my earlier comment from 09/May/14 11:42 PM. Unpatched 1.554.3 doesn't kill 'sleep' on the slave, patched does.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/scm_issue_link.png" title="scm_issue_link's avatar" /> scm_issue_link:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-22641?focusedId=206207&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-206207">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Code changed in jenkins
User: christ66
Path:
 core/src/main/java/hudson/util/ProcessTree.java
<a href="http://jenkins-ci.org/commit/jenkins/410f06adfa798d29118c77ed01c5c02fc207cb02" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins-ci.org/commit/jenkins/410f06adfa798d29118c77ed01c5c02fc207cb02</a>
Log:
  <span class="error">&#91;FIXED JENKINS-22641&#93;</span> FileUtils.readFileToByteArray behavior has changed in the latest version of commons-io.</p>


</pre>
</details>

</ul>
</details>

<p>Code changed in jenkins<br/>
User: christ66<br/>
Path:<br/>
 core/src/main/java/hudson/util/ProcessTree.java<br/>
<a href="http://jenkins-ci.org/commit/jenkins/410f06adfa798d29118c77ed01c5c02fc207cb02" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins-ci.org/commit/jenkins/410f06adfa798d29118c77ed01c5c02fc207cb02</a><br/>
Log:<br/>
  <span class="error">&#91;FIXED JENKINS-22641&#93;</span> FileUtils.readFileToByteArray behavior has changed in the latest version of commons-io.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/scm_issue_link.png" title="scm_issue_link's avatar" /> scm_issue_link:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-22641?focusedId=206208&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-206208">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Code changed in jenkins
User: christ66
Path:
 test/src/test/java/hudson/util/ProcessTreeKillerTest.java
<a href="http://jenkins-ci.org/commit/jenkins/0d4c0cb6b274bfa18c810fcf761e3ad8b27ceb34" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins-ci.org/commit/jenkins/0d4c0cb6b274bfa18c810fcf761e3ad8b27ceb34</a>
Log:
  Add test unit for <a href="https://issues.jenkins.io/browse/JENKINS-22641" title="Jenkins no longer kills running processes after job fails" class="issue-link" data-issue-key="JENKINS-22641"><del>JENKINS-22641</del></a></p>


</pre>
</details>

</ul>
</details>

<p>Code changed in jenkins<br/>
User: christ66<br/>
Path:<br/>
 test/src/test/java/hudson/util/ProcessTreeKillerTest.java<br/>
<a href="http://jenkins-ci.org/commit/jenkins/0d4c0cb6b274bfa18c810fcf761e3ad8b27ceb34" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins-ci.org/commit/jenkins/0d4c0cb6b274bfa18c810fcf761e3ad8b27ceb34</a><br/>
Log:<br/>
  Add test unit for <a href="https://issues.jenkins.io/browse/JENKINS-22641" title="Jenkins no longer kills running processes after job fails" class="issue-link" data-issue-key="JENKINS-22641"><del>JENKINS-22641</del></a></p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/scm_issue_link.png" title="scm_issue_link's avatar" /> scm_issue_link:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-22641?focusedId=206209&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-206209">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Code changed in jenkins
User: christ66
Path:
 core/src/main/java/hudson/util/ProcessTree.java
<a href="http://jenkins-ci.org/commit/jenkins/9ac68b92dcc0bc093c7983cdb0aab72342165df4" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins-ci.org/commit/jenkins/9ac68b92dcc0bc093c7983cdb0aab72342165df4</a>
Log:
  Merge branch '<a href="https://issues.jenkins.io/browse/JENKINS-22641" title="Jenkins no longer kills running processes after job fails" class="issue-link" data-issue-key="JENKINS-22641"><del>JENKINS-22641</del></a>' of github.com:christ66/jenkins into <a href="https://issues.jenkins.io/browse/JENKINS-22641" title="Jenkins no longer kills running processes after job fails" class="issue-link" data-issue-key="JENKINS-22641"><del>JENKINS-22641</del></a></p>


</pre>
</details>

</ul>
</details>

<p>Code changed in jenkins<br/>
User: christ66<br/>
Path:<br/>
 core/src/main/java/hudson/util/ProcessTree.java<br/>
<a href="http://jenkins-ci.org/commit/jenkins/9ac68b92dcc0bc093c7983cdb0aab72342165df4" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins-ci.org/commit/jenkins/9ac68b92dcc0bc093c7983cdb0aab72342165df4</a><br/>
Log:<br/>
  Merge branch '<a href="https://issues.jenkins.io/browse/JENKINS-22641" title="Jenkins no longer kills running processes after job fails" class="issue-link" data-issue-key="JENKINS-22641"><del>JENKINS-22641</del></a>' of github.com:christ66/jenkins into <a href="https://issues.jenkins.io/browse/JENKINS-22641" title="Jenkins no longer kills running processes after job fails" class="issue-link" data-issue-key="JENKINS-22641"><del>JENKINS-22641</del></a></p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/scm_issue_link.png" title="scm_issue_link's avatar" /> scm_issue_link:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-22641?focusedId=206210&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-206210">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Code changed in jenkins
User: Jesse Glick
Path:
 core/src/main/java/hudson/util/ProcessTree.java
 test/src/test/java/hudson/util/ProcessTreeKillerTest.java
<a href="http://jenkins-ci.org/commit/jenkins/f03e67a49bf01343635c60e894b9617483a03011" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins-ci.org/commit/jenkins/f03e67a49bf01343635c60e894b9617483a03011</a>
Log:
  Merge branch '<a href="https://issues.jenkins.io/browse/JENKINS-22641" title="Jenkins no longer kills running processes after job fails" class="issue-link" data-issue-key="JENKINS-22641"><del>JENKINS-22641</del></a>' of github.com:christ66/jenkins</p>


</pre>
</details>

</ul>
</details>

<p>Code changed in jenkins<br/>
User: Jesse Glick<br/>
Path:<br/>
 core/src/main/java/hudson/util/ProcessTree.java<br/>
 test/src/test/java/hudson/util/ProcessTreeKillerTest.java<br/>
<a href="http://jenkins-ci.org/commit/jenkins/f03e67a49bf01343635c60e894b9617483a03011" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins-ci.org/commit/jenkins/f03e67a49bf01343635c60e894b9617483a03011</a><br/>
Log:<br/>
  Merge branch '<a href="https://issues.jenkins.io/browse/JENKINS-22641" title="Jenkins no longer kills running processes after job fails" class="issue-link" data-issue-key="JENKINS-22641"><del>JENKINS-22641</del></a>' of github.com:christ66/jenkins</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/scm_issue_link.png" title="scm_issue_link's avatar" /> scm_issue_link:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-22641?focusedId=206211&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-206211">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Code changed in jenkins
User: Jesse Glick
Path:
 changelog.html
<a href="http://jenkins-ci.org/commit/jenkins/14a147eda8eebfd78c1cc53e1d59232d720de6b7" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins-ci.org/commit/jenkins/14a147eda8eebfd78c1cc53e1d59232d720de6b7</a>
Log:
  <a href="https://issues.jenkins.io/browse/JENKINS-22641" title="Jenkins no longer kills running processes after job fails" class="issue-link" data-issue-key="JENKINS-22641"><del>JENKINS-22641</del></a> Noting merge of #1322.</p>


<p>Compare: <a href="https://github.com/jenkinsci/jenkins/compare/a8756c6c0ddc...14a147eda8ee" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/jenkins/compare/a8756c6c0ddc...14a147eda8ee</a></p>

</pre>
</details>

</ul>
</details>

<p>Code changed in jenkins<br/>
User: Jesse Glick<br/>
Path:<br/>
 changelog.html<br/>
<a href="http://jenkins-ci.org/commit/jenkins/14a147eda8eebfd78c1cc53e1d59232d720de6b7" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins-ci.org/commit/jenkins/14a147eda8eebfd78c1cc53e1d59232d720de6b7</a><br/>
Log:<br/>
  <a href="https://issues.jenkins.io/browse/JENKINS-22641" title="Jenkins no longer kills running processes after job fails" class="issue-link" data-issue-key="JENKINS-22641"><del>JENKINS-22641</del></a> Noting merge of #1322.</p>


<p>Compare: <a href="https://github.com/jenkinsci/jenkins/compare/a8756c6c0ddc...14a147eda8ee" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/jenkins/compare/a8756c6c0ddc...14a147eda8ee</a></p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/dogfood.png" title="dogfood's avatar" /> dogfood:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-22641?focusedId=206214&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-206214">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Integrated in <span class="image-wrap" style=""><img src="http://ci.jenkins-ci.org/images/16x16/blue.png" style="border: 0px solid black" /></span> <a href="http://ci.jenkins-ci.org/job/jenkins_main_trunk/3538/" class="external-link" target="_blank" rel="nofollow noopener">jenkins_main_trunk #3538</a>
     <span class="error">&#91;FIXED JENKINS-22641&#93;</span> FileUtils.readFileToByteArray behavior has changed in the latest version of commons-io. (Revision 410f06adfa798d29118c77ed01c5c02fc207cb02)
Add test unit for <a href="https://issues.jenkins.io/browse/JENKINS-22641" title="Jenkins no longer kills running processes after job fails" class="issue-link" data-issue-key="JENKINS-22641"><del>JENKINS-22641</del></a> (Revision 0d4c0cb6b274bfa18c810fcf761e3ad8b27ceb34)
<a href="https://issues.jenkins.io/browse/JENKINS-22641" title="Jenkins no longer kills running processes after job fails" class="issue-link" data-issue-key="JENKINS-22641"><del>JENKINS-22641</del></a> Noting merge of #1322. (Revision 14a147eda8eebfd78c1cc53e1d59232d720de6b7)</p>

<p>     Result = SUCCESS
schristou88 : <a href="https://github.com/jenkinsci/jenkins/commit/410f06adfa798d29118c77ed01c5c02fc207cb02" class="external-link" target="_blank" rel="nofollow noopener">410f06adfa798d29118c77ed01c5c02fc207cb02</a>
Files : </p>
<ul>
	<li>core/src/main/java/hudson/util/ProcessTree.java</li>
</ul>


<p>schristou88 : <a href="https://github.com/jenkinsci/jenkins/commit/0d4c0cb6b274bfa18c810fcf761e3ad8b27ceb34" class="external-link" target="_blank" rel="nofollow noopener">0d4c0cb6b274bfa18c810fcf761e3ad8b27ceb34</a>
Files : </p>
<ul>
	<li>test/src/test/java/hudson/util/ProcessTreeKillerTest.java</li>
</ul>


<p>Jesse Glick : <a href="https://github.com/jenkinsci/jenkins/commit/14a147eda8eebfd78c1cc53e1d59232d720de6b7" class="external-link" target="_blank" rel="nofollow noopener">14a147eda8eebfd78c1cc53e1d59232d720de6b7</a>
Files : </p>
<ul>
	<li>changelog.html</li>
</ul>

</pre>
</details>

</ul>
</details>

<p>Integrated in <span class="image-wrap" style=""><img src="http://ci.jenkins-ci.org/images/16x16/blue.png" style="border: 0px solid black" /></span> <a href="http://ci.jenkins-ci.org/job/jenkins_main_trunk/3538/" class="external-link" target="_blank" rel="nofollow noopener">jenkins_main_trunk #3538</a><br/>
     <span class="error">&#91;FIXED JENKINS-22641&#93;</span> FileUtils.readFileToByteArray behavior has changed in the latest version of commons-io. (Revision 410f06adfa798d29118c77ed01c5c02fc207cb02)<br/>
Add test unit for <a href="https://issues.jenkins.io/browse/JENKINS-22641" title="Jenkins no longer kills running processes after job fails" class="issue-link" data-issue-key="JENKINS-22641"><del>JENKINS-22641</del></a> (Revision 0d4c0cb6b274bfa18c810fcf761e3ad8b27ceb34)<br/>
<a href="https://issues.jenkins.io/browse/JENKINS-22641" title="Jenkins no longer kills running processes after job fails" class="issue-link" data-issue-key="JENKINS-22641"><del>JENKINS-22641</del></a> Noting merge of #1322. (Revision 14a147eda8eebfd78c1cc53e1d59232d720de6b7)</p>

<p>     Result = SUCCESS<br/>
schristou88 : <a href="https://github.com/jenkinsci/jenkins/commit/410f06adfa798d29118c77ed01c5c02fc207cb02" class="external-link" target="_blank" rel="nofollow noopener">410f06adfa798d29118c77ed01c5c02fc207cb02</a><br/>
Files : </p>
<ul>
	<li>core/src/main/java/hudson/util/ProcessTree.java</li>
</ul>


<p>schristou88 : <a href="https://github.com/jenkinsci/jenkins/commit/0d4c0cb6b274bfa18c810fcf761e3ad8b27ceb34" class="external-link" target="_blank" rel="nofollow noopener">0d4c0cb6b274bfa18c810fcf761e3ad8b27ceb34</a><br/>
Files : </p>
<ul>
	<li>test/src/test/java/hudson/util/ProcessTreeKillerTest.java</li>
</ul>


<p>Jesse Glick : <a href="https://github.com/jenkinsci/jenkins/commit/14a147eda8eebfd78c1cc53e1d59232d720de6b7" class="external-link" target="_blank" rel="nofollow noopener">14a147eda8eebfd78c1cc53e1d59232d720de6b7</a><br/>
Files : </p>
<ul>
	<li>changelog.html</li>
</ul>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/scm_issue_link.png" title="scm_issue_link's avatar" /> scm_issue_link:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-22641?focusedId=207001&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-207001">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Code changed in jenkins
User: Kohsuke Kawaguchi
Path:
 aggregator/src/test/java/org/jenkinsci/plugins/workflow/steps/durable_task/ShellStepTest.java
<a href="http://jenkins-ci.org/commit/workflow-plugin/e74fd8349c15568354fc3934010713503d22761f" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins-ci.org/commit/workflow-plugin/e74fd8349c15568354fc3934010713503d22761f</a>
Log:
  added a test bug this is blocked by <a href="https://issues.jenkins.io/browse/JENKINS-22641" title="Jenkins no longer kills running processes after job fails" class="issue-link" data-issue-key="JENKINS-22641"><del>JENKINS-22641</del></a></p>


</pre>
</details>

</ul>
</details>

<p>Code changed in jenkins<br/>
User: Kohsuke Kawaguchi<br/>
Path:<br/>
 aggregator/src/test/java/org/jenkinsci/plugins/workflow/steps/durable_task/ShellStepTest.java<br/>
<a href="http://jenkins-ci.org/commit/workflow-plugin/e74fd8349c15568354fc3934010713503d22761f" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins-ci.org/commit/workflow-plugin/e74fd8349c15568354fc3934010713503d22761f</a><br/>
Log:<br/>
  added a test bug this is blocked by <a href="https://issues.jenkins.io/browse/JENKINS-22641" title="Jenkins no longer kills running processes after job fails" class="issue-link" data-issue-key="JENKINS-22641"><del>JENKINS-22641</del></a></p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>dbogardus:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-22641?focusedId=217577&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-217577">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>This issue has returned as of version 1.587 (through current version - 1.593)</p>

<p>It was fixed in 1.565.3 through 1.586</p>

<p>Same behavior as original bug, leaves processes running when job is killed. </p>
</pre>
</details>

</ul>
</details>

<p>This issue has returned as of version 1.587 (through current version - 1.593)</p>

<p>It was fixed in 1.565.3 through 1.586</p>

<p>Same behavior as original bug, leaves processes running when job is killed. </p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/jglick.png" title="jglick's avatar" /> jglick:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-22641?focusedId=217602&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-217602">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=dbogardus" class="user-hover" rel="dbogardus">dbogardus</a> then your issue is probably a distinct bug with a similar symptom but potentially distinct preconditions and root cause. Better to file it as a new ticket, with any steps to reproduce you can muster, and mark it as “blocking” this one.</p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=dbogardus" class="user-hover" rel="dbogardus">dbogardus</a> then your issue is probably a distinct bug with a similar symptom but potentially distinct preconditions and root cause. Better to file it as a new ticket, with any steps to reproduce you can muster, and mark it as “blocking” this one.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/scm_issue_link.png" title="scm_issue_link's avatar" /> scm_issue_link:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-22641?focusedId=226117&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-226117">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Code changed in jenkins
User: Jesse Glick
Path:
 pom.xml
 src/test/java/org/jenkinsci/plugins/durabletask/BourneShellScriptTest.java
<a href="http://jenkins-ci.org/commit/durable-task-plugin/fc48f447763cb69d6ebf7c67e54e476d9e903bbd" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins-ci.org/commit/durable-task-plugin/fc48f447763cb69d6ebf7c67e54e476d9e903bbd</a>
Log:
  Added test for stop.
<a href="https://issues.jenkins.io/browse/JENKINS-22641" title="Jenkins no longer kills running processes after job fails" class="issue-link" data-issue-key="JENKINS-22641"><del>JENKINS-22641</del></a> means that this does not work in 1.554.3, so need to bump up the dependency to 1.565.3.</p>


</pre>
</details>

</ul>
</details>

<p>Code changed in jenkins<br/>
User: Jesse Glick<br/>
Path:<br/>
 pom.xml<br/>
 src/test/java/org/jenkinsci/plugins/durabletask/BourneShellScriptTest.java<br/>
<a href="http://jenkins-ci.org/commit/durable-task-plugin/fc48f447763cb69d6ebf7c67e54e476d9e903bbd" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins-ci.org/commit/durable-task-plugin/fc48f447763cb69d6ebf7c67e54e476d9e903bbd</a><br/>
Log:<br/>
  Added test for stop.<br/>
<a href="https://issues.jenkins.io/browse/JENKINS-22641" title="Jenkins no longer kills running processes after job fails" class="issue-link" data-issue-key="JENKINS-22641"><del>JENKINS-22641</del></a> means that this does not work in 1.554.3, so need to bump up the dependency to 1.565.3.</p>
